src/ai/backend/common/events.py

```diff
@@ -1159,76 +1153,59 @@ class EventDispatcher(aobject):
-        stream_key: str = "events",
-        node_id: str | None = None,
         consumer_exception_handler: AsyncExceptionHandler | None = None,
         subscriber_exception_handler: AsyncExceptionHandler | None = None,
         event_observer: EventObserver = NopEventObserver(),
     ) -> None:
-        _redis_config = redis_config.copy()
-        if service_name:
-            _redis_config["service_name"] = service_name
-        self.redis_client = redis_helper.get_redis_object(
-            _redis_config, name="event_dispatcher.stream", db=db
-        )
         self._log_events = log_events
         self._closed = False
-        self.consumers = defaultdict(set)
-        self.subscribers = defaultdict(set)
-        self._stream_key = stream_key
-        self._consumer_group = consumer_group
-        self._consumer_name = _generate_consumer_id(node_id)
+        self._consumers = defaultdict(set)
+        self._subscribers = defaultdict(set)
+        self._msg_queue = message_queue
         self._metric_observer = event_observer
-        self.consumer_taskgroup = PersistentTaskGroup(
+        self._consumer_taskgroup = PersistentTaskGroup(
             name="consumer_taskgroup",
             exception_handler=consumer_exception_handler,
         )
-        self.subscriber_taskgroup = PersistentTaskGroup(
+        self._subscriber_taskgroup = PersistentTaskGroup(
             name="subscriber_taskgroup",
             exception_handler=subscriber_exception_handler,
         )
-
-    async def __ainit__(self) -> None:
-        self.consumer_loop_task = asyncio.create_task(self._consume_loop())
-        self.subscriber_loop_task = asyncio.create_task(self._subscribe_loop())
+        self._consumer_loop_task = asyncio.create_task(self._consume_loop())
+        self._subscriber_loop_task = asyncio.create_task(self._subscribe_loop())
 
     async def close(self) -> None:
         self._closed = True
         try:
             cancelled_tasks = []
-            await self.consumer_taskgroup.shutdown()
-            await self.subscriber_taskgroup.shutdown()
-            if not self.consumer_loop_task.done():
-                self.consumer_loop_task.cancel()
-                cancelled_tasks.append(self.consumer_loop_task)
-            if not self.subscriber_loop_task.done():
-                self.subscriber_loop_task.cancel()
-                cancelled_tasks.append(self.subscriber_loop_task)
+            await self._consumer_taskgroup.shutdown()
+            await self._subscriber_taskgroup.shutdown()
+            if not self._consumer_loop_task.done():
+                self._consumer_loop_task.cancel()
+                cancelled_tasks.append(self._consumer_loop_task)
+            if not self._subscriber_loop_task.done():
+                self._subscriber_loop_task.cancel()
+                cancelled_tasks.append(self._subscriber_loop_task)
             await asyncio.gather(*cancelled_tasks, return_exceptions=True)
```

Comment:
Is it okay not to handle returned exceptions here?

Reply:
I left the existing logic untouched.
In general, the `EventDispatcher` will close when the server shuts down, so this shouldn't be a problem under normal circumstances.